### PR TITLE
Improve the error logging when SAN matching fails. Fixes #363

### LINKED ISSUE #363: Confusing error message when the cert SAN has a typo in it
**Describe the bug**

Given a cert with a SAN with a typo (note .cm not .com here), I get the following logging:

```
[DBG] check CN                   foo.example.com
[DBG] COMMON_NAME              = 192.168.0.1 foo.example.com foo
[DBG] ALTNAMES                 = 1
[DBG] SUBJECT_ALTERNATIVE_NAME = foo.example.cm foo 192.168.0.1
CRITICAL error: invalid CN ('foo.example.com' does not match 'foo.example.com, foo')
```

Whereas I'd expect/desire:
`CRITICAL error: invalid CN ('foo.example.cm' does not match 'foo.example.com, foo')`

Or certainly something that doesn't say "a does not match a" when it clearly does!

I've looked at the code a bit and I feel like I'm missing something obvious but can't see a clear fix, maybe because it's checking by the COMMON_NAME field against the SUBJECT_ALTERNATIVE_NAME rather than the other way round.

I don't know if this has arisen from #355 ?

**To Reproduce**

Have a cert with a typo and check for the valid version.

**Expected behavior**

Report the SAN which doesn't match

**System (please complete the following information):**

 - OS: Ubuntu
 - OS version: [e.g. 22]
 - check_ssl_cert version: latest
 - OpenSSL version (```openssl version```): [e.g., 3.0.0]

### PR DESCRIPTION
**Pull request by peternewman**
Fixes #363

### CODE REVIEW

File: check_ssl_cert [hunk 1]
```diff
@@ -4884,7 +4886,7 @@ main() {
                 fi
 
                 if [ -n "${fail}" ]; then
-                    prepend_critical_message "invalid CN ('${CN_TMP}' does not match '${fail}')"
+                    prepend_critical_message "invalid CN ('${fail}' not found in '${CN_TMP}')"
                 else
                     if [ -z "${ok}" ] && [ -z "${COMMON_NAME_IS_AN_IP}" ]; then
                         prepend_critical_message "invalid CN ('${CN_TMP}' does not match '${COMMON_NAME}')"
```
Comment (peternewman) [inline check_ssl_cert:4892]: I don't know if this wants updating too, I couldn't trigger that branch.

File: check_ssl_cert [hunk 2]
```diff
@@ -4803,6 +4803,8 @@ main() {
             CN_TMP="$(echo "${CN}" | sed "s/|/ PIPE /g")"
 
             if [ -n "${SUBJECT_ALTERNATIVE_NAME}" ]; then
+            
+                CN_TMP="$(echo "${SUBJECT_ALTERNATIVE_NAME}" | sed "s/|/ PIPE /g")"
```
Comment (peternewman) [inline check_ssl_cert:4807]: Maybe this should just be updated a few lines above?

Comment (matteocorti) [reply check_ssl_cert:4807]: Although a SAN should always be there, there are some certificates which don't have one.

Comment (peternewman) [reply check_ssl_cert:4807]: Of course I've got a root cert exactly like this! 🤦

Although in testing it I found another bug, fixed in #371 !

**Pull request merged**